### ISSUE TITLE
Update guidelines to allow `internal` directory in the library source structure

### ISSUE THREAD

**Issue by antkmsft** [Azure/azure-sdk-for-cpp#5091]
`samples/internal` was chosen because the hierarchy of folders in the repo is:

- `sdk/<service>/<package>`
  - inc
  - samples
  - src
  - tests
  - vcpkg
 
And who knows - there might be samples for AMQP message or AMQP value created.

_Originally posted by @LarryOsterman in https://github.com/Azure/azure-sdk-for-cpp/pull/5079#discussion_r1376469264_

In short, I think internal samples should live under `/internal/samples` and not under `samples/internal`.

**Comment by LarryOsterman** [comment]
QQ: Could you please go into more details about why we should change the guidance for package layout?

**Comment by antkmsft** [comment]
So that everything inside `samples` is "safe" for use by our customers.

**Comment by LarryOsterman** [comment]
I strongly disagree with this change. First off, all of the existing SDKs which have separate samples follow this pattern, its embedded in the [common core guidelines](https://azure.github.io/azure-sdk/policies_repostructure.html#sdk-directory-layout). And I am loathe to change our guidelines to differ from them just because the majority of the AMQP API surface is internal.
